src/dreyfus_index.erl

```diff
@@ -157,9 +159,28 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult}, State) ->
+    #state{index=#index{sig=CurrentSig}, waiting_list = WaitList} = State,
+    case check_if_index_is_deleted(DDocResult, CurrentSig) of
+        true ->
+            case WaitList of
+                [] ->
+                    self() ! index_deleted;
+                [_Index] ->
+                    erlang:send_after(?INDEX_SHUTDOWN_DELAY, self(), index_deleted)
```

Comment:
it would be better to wait until the list is empty. New requests should not be possible if the index has been deleted.